packages/sdk-v2/src/sdk/savings/react.ts

```diff
@@ -0,0 +1,216 @@
+import { useGetContract } from '../base/react'
+import { ethers } from 'ethers'
+import { EnvKey } from '../base'
+import { GoodDollarStaking, IGoodDollar } from '@gooddollar/goodprotocol/types'
+import { Currency, CurrencyAmount } from '@uniswap/sdk-core'
+import { G$ } from '@gooddollar/web3sdk'
+import { getGasPrice } from '../constants'
+
+export interface StakerInfo {
+  claimable: {
+    g$Reward: CurrencyAmount<Currency> | any,
+    goodReward: CurrencyAmount<Currency> | any
+  } | undefined, 
+  balance?: CurrencyAmount<Currency> | any,
+  rewardsPaid?: {
+    g$Minted: CurrencyAmount<Currency> | any,
+    goodMinted: CurrencyAmount<Currency> | any,
+  }
+  deposit?: CurrencyAmount<Currency> | any,
+  principle?: CurrencyAmount<Currency> | any
+  shares?: number | any,
+  lastSharePrice?: number | any
+}
+
+export interface GlobalStats {
+  totalStaked: CurrencyAmount<Currency> | undefined,
+  totalRewardsPaid: CurrencyAmount<Currency> | undefined,
+  apy: number | undefined
+  lastUpdateBlock?: number,
+  savings?: number,
+}
+
+export function useSavingsBalance(refresh: QueryParams["refresh"] = "never", env?: EnvKey) {
+  const { account } = useEthers()
+  const gooddollar = useGetContract("GoodDollar", true, "savings", env) as IGoodDollar; 
+  const gdStaking = useGetContract("GoodDollarStaking", true, "savings", env) as GoodDollarStaking;
+
+  const results = useCalls(
+    [
+      {
+        contract: gooddollar,
+        method: 'balanceOf',
+        args: [account]
+      },
+      {
+        contract: gdStaking,
+        method: 'getSavings',
+        args: [account]
+      }
+    ]
+  )
+
+  if (results[0]?.error || results[1]?.error){
+  }
+  let g$Balance: string = '0', savingsBalance: string = '0';
+  if (!results.includes(undefined)) {
+    const [balance] = results[0]?.value
+    const [sBalance] = results[1]?.value
+    
+    g$Balance = parseFloat( (parseInt(balance) / 1e2).toString() ).toFixed(2)
+    savingsBalance = parseFloat( (parseInt(sBalance) / 1e2).toString()).toFixed(2)
+  }
+  
+  return { g$Balance: g$Balance, savingsBalance: savingsBalance }
+}
+
+
+
+
+export const useSavingsFunctions = (chainId: number, env?: EnvKey) => {
+  const gooddollar = useGetContract("GoodDollar", true, "savings", env) as IGoodDollar;
+  const gdStaking = useGetContract("GoodDollarStaking", true, "savings", env) as GoodDollarStaking;
+
+  const {state: transferState, send: sendTransfer} = useContractFunction(gooddollar, "transferAndCall", {transactionName: "Transfer to savings"});
+  const {state: withdrawState, send: sendWithdraw} = useContractFunction(gdStaking, "withdrawStake", {transactionName: "Withdraw from savings"});
+  const {state: claimState, send: sendClaim} = useContractFunction(gdStaking, "withdrawRewards", {transactionName: 'Withdraw rewards from savings'});
+
+  let overrides = {}
+  if (chainId === 122){
+    overrides['gasPrice'] = getGasPrice[chainId]
+  }
```

Comment:
lets try to do this at the provider sdk level
so we dont need to this gasoverride for every contract call.
in the basesdk we can override the provider/signer getGasPrice method.
ie
signer.getGasPrice = () => chainIdToGasPrice[sdkChainId]

Reply:
I have now added the overrides for .sendTransaction (type tx) and .getGasPrice for noBaseFee chains
this only is working for when the class functions of the SDK are used which directly uses ethers.

For the react hooks of useDapp we need to figure out an alternative approach.

1. set signer from UI and add the override functions there and add the signer as config options (refs: https://github.com/TrueFiEng/useDApp/blob/9b310cb9cb2065d540d2808bd67fcf7d0eb7d241/packages/core/src/hooks/useContractFunction.ts#L93 (send function) 
https://github.com/TrueFiEng/useDApp/blob/9b310cb9cb2065d540d2808bd67fcf7d0eb7d241/packages/core/src/helpers/getSignerFromOptions.ts#L7 (getting signer)
2. override the signer in the EIPProvider from blocknative
3. still manually add the overrides to any state-changing function call when using the react-hooks
4. ?